lib/rest2.js

```diff
@@ -1407,24 +1407,45 @@ class RESTv2 {
       .then(_takeResNotify)
   }
 
+/**
+ * @typedef {object} MultiOrderOpPayload
+ * @property {number|number[]} [id] - array of order IDs or single order ID
+ */
+
+/**
+ * @typedef {Array} MultiOrderOp
+ * @property {string} 0 - operation, i.e. 'oc', 'on', 'oc_multi', 'ou'
+ * @property {MultiOrderOpPayload|Order} 1 - payload, i.e. { id: [1, 2] }
+ */
+
   /**
    * Send Multiple order-related operations.
    *
-   * @param {object} params - Multiple order-related operations parameters
-   * @param {Array.<Array.<string,object>>} params.ops - List of operations to perform
-   * @param {Array.<string,Order>} [params.ops[0]] - Order Submit ['on', new Order({...})]
-   * @param {Array.<string,{id:number}>} [params.ops[1]] - Order cancellation ['oc', { id: 1 }]
-   * @param {Array.<string,{id:Array.<number>}>} [params.ops[2]] - Multi Order Cancellation ["oc_multi", {id: [1, 2, ...]}]
-   * @param {Array.<string,Order>} [params.ops[3]] - Order Update ["ou", {id: 1, price: '15', amount: '0.001'}]
-   * @param {Function} [cb] - callback
-   * @returns {Promise} - p
+   * @param {MultiOrderOp[]} ops - array of order operations
+   * @returns {Promise} p
    * @see https://docs.bitfinex.com/reference#rest-auth-order-multi
    */
-  orderMultiOp (params = { ops: [] }, cb) {
-    params.ops = params.ops.map((op) => {
+  orderMultiOp (ops = [], cb) {
+    if (!Array.isArray(ops)) {
+      const e = new Error('ops should be an array')
+      return this._cb(e, null, cb)
+    }
+
+    if (ops.some((op) => !Array.isArray(op))) {
+      const e = new Error('ops should contain only arrays')
+      return this._cb(e, null, cb)
+    }
+
+    ops = ops.map((op) => {
       if (op[0] === 'on' && op[1]) {
-        // generate new packet from instance of Order class to ensure consistent payload signature
-        const packet = (op[1] instanceof Order ? op[1] : new Order(op[1])).toNewOrderPacket()
+        // to ensure consistent payload signature
+        // generate new packet from instance of Order class
+        const packet = (
+          op[1] instanceof Order
+            ? op[1]
+            : new Order(op[1])
+        ).toNewOrderPacket()
+
         if (this._affCode) {
           packet.meta.aff_code = packet.meta.aff_code || this._affCode
```

Comment:
`meta` here may not exist if it was not provided on any of the order packets, we **or** it in the new order packet response (`const meta = { ...(this.meta || {}) }`)

Reply:
https://github.com/bitfinexcom/bfx-api-node-models/blob/master/lib/order.js#L634 if I understand it correctly, meta will always be there at least as empty object literal after calling `toNewOrderPacket` on class instance no?

Reply:
```js
const o1 = new Order({ amount: 0.001, price: 115 })
console.log(o1.toNewOrderPacket())
```
```js
{
  gid: undefined,
  cid: 1591165278327,
  symbol: undefined,
  type: undefined,
  amount: '0.00100000',
  flags: 0,
  meta: {},
  tif: undefined,
  price: '115.00'
}
```
so
```js
if (!packet.meta) {
  packet.meta = {}
}
```
becomes redundant. Or there's some implicit case I'm not seeing that can actually remove it from packet?

Reply:
Aye good point, I passed over that. Approved :+1: